### ISSUE TITLE
Flux ping manpage fixups?

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#854]
Saw this in the	flux ping manpage:

```
-r, --rank=NODESET
           Find target on a specific broker rank(s). Default: send to FLUX_NODEID_ANY.
```

So I tried:

```
$ flux ping --rank=FLUX_NODEID_ANY cmb
flux-ping: flux_rpc: Invalid argument
```

Thinking I could actually use the string.  Going through the code, it doesn't appear to be the case.

Is this something we wanted to support? Or should the manpage just not mention it?  I assumed the latter, but then I noticed in ```flux_rpc_multi``` and ```flux-ping.c``` that we do appear to handle some special case strings.

```
    if (!strcmp (nodeset, "all")) {
	if (flux_get_size (h, &count) < 0)
            goto error;
        ns = nodeset_create_range (0, count - 1);
```

```
        } else if (!strcmp (target, "all")) {
            ctx.rank = target;
            target = "cmb";
	}
```

But "all" wasn't documented in the manpage.

Glad to fixup things, but wasn't sure what was intended.

**Comment by chu11** [comment]
Now that I looked through some other docs and see "all" supported elsewhere, I see that should be documented.  I'll fix up the manpage.

**Comment by garlick** [comment]
To improve usability, it would be great to make a pass through the utilities and make sure there is a uniform interface for specifying a set of target ranks.

**Comment by chu11** [comment]
Just checked, "all" wasn't supported in atleast ```exec``` and ```ps```.  Time to learn lua :-)

**Comment by grondo** [comment]
Oh, sorry about that. For flux-exec `"all"` is the default, and you should be able to easily add a check for `opts.r == "all"` as a special case in `get_ranklist`.

I'm not sure `flux-ps` is even used very much, but I'm sure it will be simple to do something similar in that script.

**Comment by chu11** [comment]
Figure I'll cleanup some manpage inconsistencies while I'm at it.  Anyone have a strong preference on how to specify usage?  So far we got:

1. ```*-r, --rank*'=RANKS'```
2. ```*-r, --rank*=['all'|'nodeset']```
3. ```*-r, --rank*'=NODESET'```

I like the third and indicating "all" is a special string in the description.  But that's me.

**Comment by grondo** [comment]
I also prefer 3. and each manpage for a command that takes `-r, --rank` should have a `NODESET` section describing acceptable `NODESET` format, or we could reference a common page.

**Comment by chu11** [comment]
Ya know, I didn't realize we didn't even have a ```NODESET``` section.  Yeah, that should be written up similar to how we have ```HOSTRANGE``` in various tools.

**Comment by chu11** [comment]
fixed fix #860
